server/src/main/java/org/cloudfoundry/identity/uaa/login/LoginInfoEndpoint.java

```diff
@@ -450,12 +451,13 @@ private void updateLoginPageModel(
             boolean fieldUsernameShow,
             boolean linkCreateAccountShow
     ) {
+        Comparator<SamlIdentityProviderDefinition> sortingByLinkText = Comparator.comparing(SamlIdentityProviderDefinition::getLinkText, String.CASE_INSENSITIVE_ORDER);
         model.addAttribute(LINK_CREATE_ACCOUNT_SHOW, linkCreateAccountShow);
         model.addAttribute(FIELD_USERNAME_SHOW, fieldUsernameShow);
-        model.addAttribute(IDP_DEFINITIONS, samlIdentityProviders.values());
+        model.addAttribute(IDP_DEFINITIONS, samlIdentityProviders.values().stream().sorted(sortingByLinkText).toList());
         Map<String, String> oauthLinks = new HashMap<>();
         ofNullable(oauthIdentityProviders).orElse(emptyMap()).entrySet().stream()
-                .filter(e -> e.getValue().isShowLinkText())
+                .filter(e -> e.getValue().isShowLinkText() && e.getValue() != null && e.getKey() != null)
```

Comment:
The condition needs to be rewritten as follows. The null check should happen before using the value.

e.getValue() != null && e.getValue().isShowLinkText() && e.getKey() != null

Reply:
Thanks